arch/x86/core/fatal.c

```diff
@@ -432,3 +432,18 @@ void z_x86_do_kernel_oops(const z_arch_esf_t *esf)
 
 	z_x86_fatal_error(reason, esf);
 }
+
+/**
+ * NMI handling
```

Comment:
unless you plan on implementing this for 32-bit as well I think we should move this to x86_64 only-code

Reply:
moved to x86_64